var/spack/repos/builtin/packages/seacas/package.py

```diff
@@ -146,19 +146,23 @@ class Seacas(CMakePackage):
     variant("faodel", default=False, description="Enable Faodel")
     variant("matio", default=True, description="Compile with matio (MatLab) support")
     variant("metis", default=False, description="Compile with METIS and ParMETIS")
+    variant("pamgen", default=False, description="Compile with pamgen")
     variant("x11", default=True, description="Compile with X11")
-
+    variant("zlib", default=False, description="Compile with zlib")
     # ###################### Dependencies ##########################
     depends_on("cmake@3.22:", when="@2023-10-24:", type="build")
     depends_on("cmake@3.17:", when="@:2023-05-30", type="build")
     depends_on("mpi", when="+mpi")
-
+    depends_on("zlib", when="+zlib")
+    depends_on("trilinos", when="+pamgen")
```

Comment:
We also need to modify the Trilinos package.py since it explicitly disables pamgen build and there is no variant available to enable pamgen.  I'm not sure of what is the best change -- I think adding yet another variant which would then allow this to be `depends_on("trilinos+pamgen", when="+pamgen")`

Then, the mpi complication comes in where we need to depend on the correct +mpi/~mpi build of Trilinos, so maybe something like:
```
    with when("+pamgen"):
        depends_on("trilinos~exodus+pamgen+mpi", when="+mpi")
        depends_on("trilinos~exodus+pamgen~mpi", when="~mpi")
```

Reply:
We should do that as part of this PR then.  It would be nice to have the changes linked in the git history.

Reply:
This should also be a part of my next commit but I was testing it locally first.